File: Stork.Core/Engine/Compilation/AST/Expr/FloatExprAST.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Stork.NET.Core.Engine.Compilation
{
    using Runtime;

    public class FloatExprAST : ExprAST
    {
        private double _value;

        public FloatExprAST(double value)
        {
            _value = value;
        }

        public double GetValue()
        {
            return _value;
        }

        #region ExprAST Members

        public override Expr Translate(Translator translator)
        {
            return translator.Translate(this);
        }

        public override Expr Assign(Translator translator, ExprAST value)
        {
            return translator.Assign(this, value);
        }

        public override Type TypeOf(Translator translator)
        {
            return translator.TypeOf(this);
        }

        #endregion
    }
}

File: Stork.JavaImplementation/Lesson4JavaCode/sigpwned-stork-0a9c0d4/src/main/java/com/sigpwned/stork/engine/compilation/ast/expr/FloatExprAST.java
	package com.sigpwned.stork.engine.compilation.ast.expr;

import com.sigpwned.stork.engine.compilation.Translator;
import com.sigpwned.stork.engine.compilation.Type;
import com.sigpwned.stork.engine.compilation.ast.ExprAST;
import com.sigpwned.stork.engine.runtime.Expr;

public class FloatExprAST extends ExprAST {
	private double value;
	
	public FloatExprAST(double value) {
		this.value = value;
	}

	public double getValue() {
		return value;
	}

	public Expr translate(Translator translator) {
		return translator.translate(this);
	}

	public Expr assign(Translator translate, ExprAST value) {
		return translate.assign(this, value);
	}
	
	public Type typeOf(Translator translate) {
		return translate.typeOf(this);
	}
}
